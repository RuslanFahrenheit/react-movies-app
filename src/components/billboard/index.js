export { default as Billboard } from './billboard';
