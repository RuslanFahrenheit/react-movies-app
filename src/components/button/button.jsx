import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './button.scss';

const Button = ({
  children,
  component: ButtonComponent = 'button',
  onClick,
  type = 'button',
  variant = '',
  expanded = false,
  additionalClass = '',
  ...rest
}) => (
  <ButtonComponent
    className={
      classNames('button', {
        [`button--${variant}`]: variant,
        'button--expanded': expanded,
        [additionalClass]: additionalClass,
      })
    }
    type={type}
    onClick={onClick}
    {...rest}
  >
    {children}
  </ButtonComponent>
);

const {
  bool,
  string,
  element,
  node,
  func,
  oneOf,
  oneOfType,
} = PropTypes;

Button.propTypes = {
  children: oneOfType([string, element, node]).isRequired,
  component: oneOfType([string, func]),
  onClick: func,
  type: string,
  variant: oneOf(['primary', 'secondary']),
  expanded: bool,
  additionalClass: string,
};

export { Button };
