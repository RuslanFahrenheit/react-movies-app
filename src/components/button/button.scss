.button {
  padding: 5px 10px;
  font-weight: 600;
  font-size: 20px;
  font-family: inherit;
  color: #fff;
  border: none;
  background-color: transparent;
  cursor: pointer;

  &:focus {
    outline: 1px solid #fff;
  }

  &--add {
    position: relative;
    z-index: 1;
    height: 40px;

    &:before {
      content: '';
      position: absolute;
      z-index: -1;
      top: 0;
      right: 0;
      width: 100%;
      height: 100%;
      background-color: #424242;
      opacity: 0.7;
    }
  }

  &--primary {
    height: 40px;
    font-weight: 200;
    background-color: #f65261;
  }
}
