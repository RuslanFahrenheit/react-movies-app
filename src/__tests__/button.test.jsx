import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Button } from '../components/button';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render a button', () => {
  act(() => {
    render(<Button>Button</Button>, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();

  act(() => {
    render(
      <Button
        type="submit"
        additionalClass="test-class"
        expanded={true}
        variant="secondary"
        onClick={() => {
        }}
      >
        Submit Button
      </Button>,
      container,
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();

  act(() => {
    render(<Button component="span">Link</Button>, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
