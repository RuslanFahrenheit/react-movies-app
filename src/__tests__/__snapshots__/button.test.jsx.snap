// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a button 1`] = `"<button class=\\"button\\" type=\\"button\\">Button</button>"`;

exports[`should render a button 2`] = `"<button class=\\"button button--secondary button--expanded test-class\\" type=\\"submit\\">Submit Button</button>"`;

exports[`should render a button 3`] = `"<span class=\\"button\\" type=\\"button\\">Link</span>"`;
